core/selection.js

```diff
@@ -926,11 +926,16 @@
 					evt.data.preventDefault();
 				}
 			}, editor );
+
 			// Always fire the selection change on focus gain.
 			// On Webkit do this on DOMFocusIn, because the selection is unlocked on it too and
 			// we need synchronization between those listeners to not lost cached editor._.previousActive property
 			// (which is updated on selectionCheck).
-			editable.attachListener( editable, CKEDITOR.env.webkit ? 'DOMFocusIn' : 'focus', function() {
+			// On Gecko with inline editor use focusin insted focus because the last one is not bubbling and
+			// event will not be fired with nested contenteditable (https://github.com/ckeditor/ckeditor-dev/issues/1113).
+			// For IE we already are replacing focus with focusin.
+			var focusEvent = ( CKEDITOR.env.webkit && 'DOMFocusIn' ) || ( isInline && CKEDITOR.env.gecko && 'focusin' ) || 'focus';
```

Comment:
I'm wondering if it could be simplified and just use `focusin` event (`DOMFocusIn` seems like webkit-only alternative, however it should also support standard `focusin`).

Reply:
I already tried `focusin` for webkit and it works fine. The same with `DOMFocusOut` replaced to `focusout`. I think I should change `blur` to `focusout` for gecko also so it will be fired from nested editables ( it's already implemented for webkit)